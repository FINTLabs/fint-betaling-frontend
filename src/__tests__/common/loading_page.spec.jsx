import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import LoadingPage from '../../common/scaffold/loading_page';

describe('Loading page', () => {
    const initialState = {
        payment: {
            payment: {school: {}},
        },
        customers: {},
        dates: {},
        employers: {},
        groups: {},
        me: {},
        mva: {},
        orderLines: {},
        payments: {},
    };
    const mockStore = configureStore();
    it('should match snapshot', () => {
        const store = mockStore(initialState);

        const tree = renderer
            .create(<Provider store={store}><LoadingPage/></Provider>)
            .toJSON();

        expect(tree)
            .toMatchSnapshot();
    });
});
