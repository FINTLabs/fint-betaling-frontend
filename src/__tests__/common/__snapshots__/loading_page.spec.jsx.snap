// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading page should match snapshot 1`] = `
<div
  className="MuiBox-root MuiBox-root-12 makeStyles-loadingContainer-1"
>
  <img
    className="makeStyles-vigoLogo-10"
    src="vigo-logo-no-iks.svg"
    title="Vigo IKS logo"
  />
  <h5
    className="MuiTypography-root makeStyles-loadingText-2 MuiTypography-h5"
  >
    Betaling
  </h5>
  <div
    className="MuiCircularProgress-root makeStyles-progress-3 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
  <p
    className="MuiTypography-root makeStyles-loadingText-2 MuiTypography-body1"
  >
    Loading... NaN%
  </p>
  <div
    className="MuiBox-root MuiBox-root-55 makeStyles-linearProgressBox-4"
  >
    <div
      aria-valuenow={NaN}
      className="MuiLinearProgress-root MuiLinearProgress-colorSecondary makeStyles-linearProgress-5 MuiLinearProgress-determinate"
      role="progressbar"
    >
      <div
        className="MuiLinearProgress-bar MuiLinearProgress-barColorSecondary MuiLinearProgress-bar1Determinate"
        style={
          Object {
            "transform": "translateX(NaN%)",
          }
        }
      />
    </div>
    <div
      className="MuiFormControl-root makeStyles-formControl-6"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
        data-shrink={true}
        htmlFor="age-simple"
      >
        Velg skole
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <div
          aria-haspopup="listbox"
          className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
          id="select-school"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </div>
        <input
          id="school-simple"
          name="school"
          onAnimationStart={[Function]}
          type="hidden"
          value={Object {}}
        />
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root MuiSelect-icon"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
    <ul
      className="MuiList-root MuiList-padding"
    >
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
          >
            Laster elever
          </span>
        </div>
        <div
          className="MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root makeStyles-loadingIcon-7"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13H5v-2h14v2z"
            />
          </svg>
        </div>
      </li>
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
          >
            Laster ansatte
          </span>
        </div>
        <div
          className="MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root makeStyles-loadingIcon-7"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13H5v-2h14v2z"
            />
          </svg>
        </div>
      </li>
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
          >
            Laster basisgrupper og undervisningsgrupper
          </span>
        </div>
        <div
          className="MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root makeStyles-loadingIcon-7"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13H5v-2h14v2z"
            />
          </svg>
        </div>
      </li>
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
          >
            Laster personlig data
          </span>
        </div>
        <div
          className="MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root makeStyles-loadingIcon-7"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13H5v-2h14v2z"
            />
          </svg>
        </div>
      </li>
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
          >
            Laster MVA-satser
          </span>
        </div>
        <div
          className="MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root makeStyles-loadingIcon-7"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13H5v-2h14v2z"
            />
          </svg>
        </div>
      </li>
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
          >
            Laster forfallsdatoer
          </span>
        </div>
        <div
          className="MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root makeStyles-loadingIcon-7"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13H5v-2h14v2z"
            />
          </svg>
        </div>
      </li>
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
          >
            Laster produkter
          </span>
        </div>
        <div
          className="MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root makeStyles-loadingIcon-7"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13H5v-2h14v2z"
            />
          </svg>
        </div>
      </li>
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
          >
            Laster betalingshistorikk
          </span>
        </div>
        <div
          className="MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root makeStyles-loadingIcon-7"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13H5v-2h14v2z"
            />
          </svg>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
